src/docfx/build/xref/XrefMapLoader.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+
+namespace Microsoft.Docs.Build
+{
+    internal class XrefMapLoader
+    {
+        public static ListBuilder<(string, Lazy<IXrefSpec>)> Load(string filePath)
+        {
+            var result = new ListBuilder<(string, Lazy<IXrefSpec>)>();
+            var content = File.ReadAllBytes(filePath);
+
+            // TODO: cache this position mapping if xref map file not updated, reuse it
+            var xrefSpecPositions = GetXrefSpecPositions(content);
+
+            ParallelUtility.ForEach(xrefSpecPositions, item =>
+            {
+                using (var stream = File.Open(filePath, FileMode.Open, FileAccess.Read, FileShare.Read))
+                {
+                    var json = ReadJsonFragment(stream, item.start, item.end);
+                    result.Add((item.uid, new Lazy<IXrefSpec>(() => JsonUtility.Deserialize<ExternalXrefSpec>(json, filePath))));
+                }
+            });
+            return result;
+        }
+
+        private static List<(string uid, long start, long end)> GetXrefSpecPositions(ReadOnlySpan<byte> content)
+        {
+            var result = new List<(string uid, long start, long end)>();
+            var reader = new Utf8JsonReader(content, isFinalBlock: true, default);
+            string uid = null;
+            int startIndex = 0;
+            int endIndex = 0;
+            while (reader.Read())
+            {
+                switch (reader.TokenType)
+                {
+                    case JsonTokenType.PropertyName:
+                        if (reader.TextEquals(Encoding.UTF8.GetBytes("uid")) && reader.Read() && reader.TokenType == JsonTokenType.String)
```

Comment:
cache utf8 uid as a static var can avoid allocation and expensive getbytes calls